### PR TITLE
Rework updating excerpts and current view to fix the related issues

### PR DESCRIPTION
**Pull request by dmitrio95**
This pull request is intended to resolve the problems handled by #3991 and some other problems related to tracking the current score view including [some crashes](https://musescore.org/en/node/276628). These problems were related to updating excerpts if its number has changed (on score loading or as a result of adding/removing parts). In this pull request excerpts updating process was somewhat reworked regarding maintaining current score view being set and tracked correctly.

The main idea of the proposed changes is in two points:
- `ScoreTab` now decides whether it needs to update its score's excerpts so that there is no need to handle this on the main MuseScore window's level (`MuseScore` class). Previously such a handling should have been done in several places and it was done in somewhat inconsistent way.
- `MuseScore` class representing the main window now tracks the currently active `ScoreTab` (that is, the currently active half of the window in case of split view) and is responsible for the decision on setting or not setting a focus on a specific score view (that is, not blocking or blocking signals from `ScoreTab`) when there might be a need to update excerpts for some score in both `ScoreTab`s.

Such changes should make current view setting behavior more consistent and help to avoid various bugs including those listed in the commit's message.

### CODE REVIEW

File: mscore/scoretab.h [hunk 1]
```diff
@@ -32,9 +32,9 @@ enum class MagIdx : char;
 //---------------------------------------------------------
 
 struct TabScoreView {
-      Score* score;
+      MasterScore* score;
```
Comment (anatoly-os) [inline mscore/scoretab.h]: Why did you change this, whether `TabScoreView` now operates on `MasterScore's` only? Who is responsible for updating excerpts views?

File: mscore/scoretab.h [hunk 2]
```diff
@@ -52,6 +52,8 @@ class ScoreTab : public QWidget {
       QStackedLayout* stack;
       MuseScore* mainWindow;
       void clearTab2();
+      TabScoreView* tabScoreView(int idx) { return static_cast<TabScoreView*>(tab->tabData(idx).value<void*>()); }
```
Comment (anatoly-os) [inline mscore/scoretab.h]: Can we avoid `static_cast` conversions here? What could be done to avoid this?

File: mscore/scoretab.h (hunk 1 above)
Comment (dmitrio95) [reply mscore/scoretab.h]: In fact, every score that was stored in this structure was a `MasterScore` since [the only place this structure's instances were created](https://github.com/musescore/MuseScore/blob/4899a7cd69cd6594351e10114ef039d6bb79707a/mscore/scoretab.cpp#L352) is an `ScoreTab::insertTab` function which accepts `MasterScore`s only. I changed this line to further ensure it and avoid type casts when [using `MasterScore::excerptsChanged`](https://github.com/musescore/MuseScore/pull/4004/files#diff-1baba45712fb8c496822b81e84988e48R268) function on the stored score.
Excerpt views are handled by `ScoreTab` either and information on the current excerpt view is represented by `part` integer field in the discussed `TabScoreView` structure.

File: mscore/scoretab.h (hunk 2 above)
Comment (dmitrio95) [reply mscore/scoretab.h]: Since an object returned stored in `QTabBar` and returned by `tabData()` is `QVariant` I believe the only way to avoid `static_cast` here is to register `TabScoreView` structure in Qt's meta-object system. Should we do it here?

**Comment by anatoly-os** [comment]
@dmitrio95 could you please rebase the PR?

**Comment by dmitrio95** [comment]
@anatoly-os Rebased it now.
